File: src/main/java/com/ada/conversation/context/PromptContextItem.java
package com.ada.conversation.context;

import com.ada.conversation.application.dto.*; import com.ada.shared.observability.MeasuredContextItem; import org.springframework.core.annotation.Order; import org.springframework.stereotype.Component;
@Component @Order(20) @MeasuredContextItem("prompt") public class PromptContextItem implements ContextItem { public LlmContentComponent component(){return LlmContentComponent.PROMPT;} public ContextState apply(ChatRequest r,ContextState c){var m=new java.util.ArrayList<>(c.messages());m.add(new LlmMessage(LlmMessageRole.USER,r.message(),component()));return new ContextState(m,c.tools());} }

File: src/main/kotlin/com/ada/conversation/context/PromptContextItem.kt
package com.ada.conversation.context

import com.ada.conversation.application.dto.ChatRequest
import com.ada.conversation.application.dto.ContextContribution
import com.ada.conversation.application.dto.LlmContentComponent
import com.ada.conversation.application.dto.LlmMessage
import com.ada.conversation.application.dto.LlmMessageRole
import org.springframework.core.annotation.Order
import org.springframework.stereotype.Component
import com.ada.shared.observability.MeasuredContextItem

@Component
@Order(20)
@MeasuredContextItem("prompt")
class PromptContextItem : ContextItem {
    override val component = LlmContentComponent.PROMPT

    override fun build(request: ChatRequest): ContextContribution = ContextContribution(
        messages = listOf(LlmMessage(LlmMessageRole.USER, request.message, component)),
    )
}
